chia/consensus/blockchain_check_conditions.py

```diff
@@ -93,23 +92,25 @@ def blockchain_assert_announcement(condition: ConditionWithArgs, announcements:
 
 def blockchain_check_conditions_dict(
     unspent: CoinRecord,
-    announcements: List[Announcement],
+    coin_announcement_names: List[bytes32],
+    puzzle_announcement_names: List[bytes32],
     conditions_dict: Dict[ConditionOpcode, List[ConditionWithArgs]],
     prev_transaction_block_height: uint32,
     timestamp: uint64,
 ) -> Optional[Err]:
     """
     Check all conditions against current state.
     """
-    announcement_names = set([a.name() for a in announcements])
     for con_list in conditions_dict.values():
         cvp: ConditionWithArgs
         for cvp in con_list:
             error = None
             if cvp.opcode is ConditionOpcode.ASSERT_MY_COIN_ID:
                 error = blockchain_assert_my_coin_id(cvp, unspent)
-            elif cvp.opcode is ConditionOpcode.ASSERT_ANNOUNCEMENT:
-                error = blockchain_assert_announcement(cvp, announcement_names)
+            elif cvp.opcode is ConditionOpcode.ASSERT_COIN_ANNOUNCEMENT:
+                error = blockchain_assert_announcement(cvp, coin_announcement_names)
+            elif cvp.opcode is ConditionOpcode.ASSERT_PUZZLE_ANNOUNCEMENT:
+                error = blockchain_assert_announcement(cvp, puzzle_announcement_names)
```

Comment:
By using the common subroutine, the error code doesn't distinguish between the two type of potentially failing announcements. Getting a failure based on the type of announcement might help someone debug a contract at some point (admittedly, they likely won't be using this code path to test it).

Reply:
(That is, consider splitting `ASSERT_ANNOUNCE_CONSUMED_FAILED` into two errors.)